### DISCUSSION TITLE
How can I efficiently force a UI refresh of individual rows?

### DISCUSSION THREAD

**Discussion by AlexanderBlackman** [w-ahmad/WinUI.TableView#82] [Q&A] [answered]
I have some bulk commands that change all selected rows at once.  
Right now, the only way I know to force a refresh is to reload the entire ItemsSource, but this isn't efficient.
There must be a better way, but changing UpdateSourceTrigger in a column's datatemplate doesn't work,  nor does UpdateLayout

``` csharp
private void HardRefreshItems() //Very wasteful
    {
        _dispatcherQueue.TryEnqueue(() =>
        {
            var items = this.ItemsSource;
            var selected = SelectedItems.Cast<TodoItem>().ToList();
            this.ItemsSource = null;
            this.ItemsSource = items;
            _updatingSelection = true;
            foreach (var item in selected.Where(i => Items.Contains(i)))
            {
                this.SelectedItems.Add(item);
            }
            _updatingSelection = false;
            _messenger.Send(message: new TodoSelectionChangedMessage(selected, fromView: true));

        });
    }

**Comment by w-ahmad** [comment]
@AlexanderBlackman, I'm afraid I couldn't understand the problem that you are facing, could you give some more information about your implementation?

**Reply by AlexanderBlackman** [reply]
Sorry, I missed your message.
The issue is normally changed rows only visually update once they've been moved away from, which is great when editing individually, but doesn't work well for operations changing many rows at once.

**Reply by w-ahmad** [reply, marked as answer]
here what I understood. 
If you want to update cell values from code behind logic then: if you have implemented `INotifyPropertyChanged` on your models then it is not difficult to update individual items from code behind.

it would be very helpful to understand issue if you can upload a full example or working application.

**Reply by AlexanderBlackman** [reply]
Thank you, I ended up doing that.
